content/en/blog/_posts/2023-04-06-seccomp-profiles-edge.md

```diff
@@ -0,0 +1,273 @@
+	}
+}
+```
+
+[uname]: https://man7.org/linux/man-pages/man2/uname.2.html
+
+Building a statically linked binary from that code can be done by:
+
+```console
+> go build -o main main.go
+> ldd ./main
+        not a dynamic executable
+```
+
+Now we can download the latest binary of [`spoc` from GitHub][spoc-latest] and
+run the application with it:
+
+[spoc-latest]: https://github.com/kubernetes-sigs/security-profiles-operator/releases/download/v0.7.1/spoc.amd64
+
+```console
+> sudo ./spoc record ./main
+10:08:25.591945 Loading bpf module
+10:08:25.591958 Using system btf file
+libbpf: loading object 'recorder.bpf.o' from buffer
+…
+libbpf: prog 'sys_enter': relo #3: patched insn #22 (ALU/ALU64) imm 16 -> 16
+10:08:25.610767 Getting bpf program sys_enter
+10:08:25.610778 Attaching bpf tracepoint
+10:08:25.611574 Getting syscalls map
+10:08:25.611582 Getting pid_mntns map
+10:08:25.613097 Module successfully loaded
+10:08:25.613311 Processing events
+10:08:25.613693 Running command with PID: 336007
+10:08:25.613835 Received event: pid: 336007, mntns: 4026531841
+10:08:25.613951 No container ID found for PID (pid=336007, mntns=4026531841, err=unable to find container ID in cgroup path)
+10:08:25.614856 Processing recorded data
+10:08:25.614975 Found process mntns 4026531841 in bpf map
+10:08:25.615110 Got syscalls: read, close, mmap, rt_sigaction, rt_sigprocmask, madvise, nanosleep, clone, uname, sigaltstack, arch_prctl, gettid, futex, sched_getaffinity, exit_group, openat
+10:08:25.615195 Adding base syscalls: access, brk, capget, capset, chdir, chmod, chown, close_range, dup2, dup3, epoll_create1, epoll_ctl, epoll_pwait, execve, faccessat2, fchdir, fchmodat, fchown, fchownat, fcntl, fstat, fstatfs, getdents64, getegid, geteuid, getgid, getpid, getppid, getuid, ioctl, keyctl, lseek, mkdirat, mknodat, mount, mprotect, munmap, newfstatat, openat2, pipe2, pivot_root, prctl, pread64, pselect6, readlink, readlinkat, rt_sigreturn, sched_yield, seccomp, set_robust_list, set_tid_address, setgid, setgroups, sethostname, setns, setresgid, setresuid, setsid, setuid, statfs, statx, symlinkat, tgkill, umask, umount2, unlinkat, unshare, write
+10:08:25.616293 Wrote seccomp profile to: /tmp/profile.yaml
+10:08:25.616298 Unloading bpf module
+```
+
+We have to execute `spoc` as root because it will internally run an [ebpf][ebpf]
+program by reusing the same code parts from the Security Profiles Operator
+itself. We can see that the bpf module got loaded successfully and `spoc`
+attached the required tracepoint to it. Then it will track the main application
+by using its [mount namespace][mntns] and process the recorded syscall data.
+
+[ebpf]: https://ebpf.io
+[mntns]: https://man7.org/linux/man-pages/man7/mount_namespaces.7.html
+
+The logs indicate that `spoc` found the syscalls `read`, `close`,
+`mmap` and so on, including `uname`. The other syscalls are coming from the
+golang runtime and its garbage collection, which already adds overhead to a
```

Comment:
I think you meant here "the other syscalls than uname that our application is calling" right?

Reply:
Yes, fixed.